### ISSUE TITLE
Returning Covariant class in Generic Protocols gives warning message

### ISSUE THREAD

**Issue by romulocollopy** [microsoft/pyright#4215]
**Describe the bug**
If we return a class instead of a bounded instance in a Generic Protocol method, we get a warning about the covariant nature of the class. Making the class invariant asses pyright check and fails mypy check.

--

According to PEP-0544, generic protocols can have methods that return an instance of a covariant type:

https://peps.python.org/pep-0544/#generic-protocols

```python
from typing import Protocol, TypeVar
T_co = TypeVar("T_co", covariant=True)


class Box(Protocol[T_co]):
    def content(self) -> T_co: ):
        ...
```

As far as I could understand from the example below in the PEP, the TypeVar cannot be a covariant if it is assigned to a mutable attribute.

```python
class Proto(Protocol[T]):
    attr: T  # this class is invariant, since it has a mutable attribute
```

If I use `Type[T]` as the return type of a method, Pyright says that it needs to be invariant. 

But if I change my code to `covariant=False`, the warning in pyright is gone but I get this error from Mypy:

```
generic_protocol.py:6: error: Invariant type variable "T" used in protocol where covariant one is expected
 [misc]
    class Box(Protocol[T]):
    ^
```

The PEP also states that:

> To distinguish between protocol class variables and protocol instance variables, the special ClassVar annotation should be used as specified by [PEP 526](https://peps.python.org/pep-0526). By default, protocol variables as defined above are considered readable and writable. To define a read-only protocol variable, one can use an (abstract) property.

And If I try to use `T` as ClassVar, I get the warning
> "ClassVar" type cannot include type variables (Pyright reportGeneralTypeIssues)

If I use the strategy suggested by the pep (below), I get no complaints from pyright, but mypy also gives me:
> error: Invariant type variable "T_co" used in protocol where covariant one is expected
 [misc] 

```python
from abc import abstractmethod
from typing import Protocol, Type, TypeVar

T_co = TypeVar("T_co", covariant=False)


class Box(Protocol[T_co]):
    @property
    @abstractmethod
    def content(self) -> Type[T_co]:
        ...
``` 

**To Reproduce**
```python
from typing import Protocol, Type, TypeVar
T_co = TypeVar("T_co", covariant=True)


class Box(Protocol[T_co]):
    def content(self) -> Type[T_co]: ):  # Type variable "T_co" used in generic protocol "Box" should be invariant (Pyright reportInvalidTypeVarUse)
        ...
```

**Expected behavior**
As the return type is not a mutable attribute, I think that Mypy behaviour's is the correct one.
No warning should be shown and a consistent behavior between mypy and pyright would be expected.

**VS Code extension or command-line**
I'm running in neovim. 
[coc.nvim] coc-pyright 1.1.280 with Pyright 1.1.280

And command line
```
myuser@2d890ee4cbf9:~$ python -m pyright -p pyproject.toml generic_protocol.py
Configuration file not found at /home/myuser/pyproject.toml.
pyproject.toml file found at /home/myuser.
Loading pyproject.toml file at /home/myuser/pyproject.toml
Pyproject file "/home/myuser/pyproject.toml" is missing "[tool.pyright]" section.
stubPath /home/myuser/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
pyright 1.1.280
/home/myuser/generic_protocol.py
  /home/myuser/generic_protocol.py:7:7 - warning: Type variable "T_co" used in generic protocol "Box" should be invariant (reportInvalidTypeVarUse)
0 errors, 1 warning, 0 informations
Completed in 0.62sec
```

**Comment by romulocollopy** [comment]
Since it's difficult to guarantee that a return as Type[T_co] is a covariant, this issue should not be valid

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.281, which I just published. It will also be included in a future release of pylance.
